src/compiler/emitter.ts

```diff
@@ -1881,13 +1860,77 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
                 }
             }
 
+            function emitCallOrNewExpressionTarget(node: Expression): CallTarget {

```

Comment:
Well the functionality below looks a lot like `emitCallWithSpread`, which calls the `emitCallTarget` you've mentioned above, and you want to now call this `emitCallTarget`, so we should find some unifying scheme.

Also, on the issue of subtlety in behavior - while the name is important, I'd rather see documentation, which we are currently lacking in. Regardless of what this is called, please add a nice JSDoc comment to point out its peculiar behavior.